storops/unity/resource/system.py

```diff
@@ -124,12 +124,32 @@ def get_snap(self, _id=None, name=None, **filters):
     def create_nas_server(self, name, sp=None, pool=None, is_repl_dst=None,
                           multi_proto=None, tenant=None):
         if sp is None:
-            sp = self.get_sp().first_item
+            sp = self._auto_balance_sp()
 
         return sp.create_nas_server(name, pool,
                                     is_repl_dst=is_repl_dst,
                                     multi_proto=multi_proto, tenant=tenant)
 
+    def _auto_balance_sp(self):
+        sp_list = self.get_sp()
+        if len(sp_list) < 2:
+            LOG.debug('SPB not present. Pick SPA to host NAS server.')
+            return sp_list.first_item
+
+        # Only need to auto pick SP when both SPA and SPB present.
+        def by_sp(sp):
+            def inner(server):
+                return server.home_sp.get_id() == sp.get_id()
+            return inner
+
+        servers = self.get_nas_server()
+        servers_0 = list(filter(by_sp(sp_list[0]), servers))
+        servers_1 = list(filter(by_sp(sp_list[1]), servers))
+        sp = sp_list[1] if len(servers_0) > len(servers_1) else sp_list[0]

```

Comment:
``` python
servers = self.get_nas_server()
spa, spb = sp_list
servers_on_spa = list(filter(by_sp(spa), servers))
sp = spb if len(servers_on_spa) * 2 > len(servers) else spa
```
Another preferred way is to add a filter in `UnityNasServerList`.  And then we could use this filter everywhere.